### PR TITLE
TT-3906 - additional logic in case of non existing cert

### PR DESCRIPTION
**Pull request by vaske**
this is workaround missing cert

### CODE REVIEW

File: gateway/mw_auth_key.go [hunk 1]
```diff
@@ -91,11 +91,12 @@ func (k *AuthKey) ProcessRequest(_ http.ResponseWriter, r *http.Request, _ inter
 
 	keyExists := false
 	var session user.SessionState
+	updateSession := false
 	if key != "" {
 		key = stripBearer(key)
 	} else if authConfig.UseCertificate && key == "" && r.TLS != nil && len(r.TLS.PeerCertificates) > 0 {
 		log.Debug("Trying to find key by client certificate")
-		certHash = certs.HexSHA256(r.TLS.PeerCertificates[0].Raw)
+		certHash = k.Spec.OrgID + certs.HexSHA256(r.TLS.PeerCertificates[0].Raw)
```
Comment (vaske) [inline gateway/mw_auth_key.go:99]: we have to concatenate orgId + hash since all cert hashes are build like that

File: gateway/mw_auth_key.go [hunk 2]
```diff
@@ -107,22 +108,31 @@ func (k *AuthKey) ProcessRequest(_ http.ResponseWriter, r *http.Request, _ inter
 	if !keyExists {
 		// fallback to search by cert
 		session, keyExists = k.CheckSessionAndIdentityForValidKey(certHash, r)
-		certHash = session.KeyID
```
Comment (vaske) [inline gateway/mw_auth_key.go:110]: this override is not necessary here otherwise in case of mTLS it will be always diff than `session.Certificate`

**Review by buger — APPROVED** [review]
👍🏻

**Review by jeffy-mathew — APPROVED** [review]

**Comment by vaske** [comment]
merged manually to 3lts

**Pull request merged**